src/app/selectors.js

```diff
@@ -36,3 +38,32 @@ export const getDecodedMacaroons = macaroons => {
   });
   return decodedMacaroons;
 };
+
+/**
+  Fetches the model list from state.
+  @param {Object} state The application state.
+  @returns {Object|Null} The list of models or null if none found.
+*/
+export const getModelList = state => {
+  if (state.juju && state.juju.models) {
+    return state.juju.models;
+  }
+  return null;
+};
+
+/**
+  Gets the model UUID from the supplied name.
+  @param {String} name The name of the model.
+  @param {Object} modelInfo The list of models.
+  @returns {Object|Null} The model UUID or null if none found.
+*/
+export const getModelUUIDByName = (name, modelInfo) => {
+  if (modelInfo) {
+    for (let uuid in modelInfo) {
```

Comment:
`filter` might be nicer.

Reply:
I want to bail out as soon as it's found so `some` would be preferred, however `modelInfo` is an Object so can't filter over it. In order to use any of the array utilities I'd have to first do something like `Object.keys().some(key => { ... })` which wouldn't be as performant in the event that there is a significant number of models.